persistence-api/src/main/scala/com/evolutiongaming/akkaeffect/persistence/Snapshotter.scala

```diff
@@ -0,0 +1,222 @@
+package com.evolutiongaming.akkaeffect.persistence
+
+import akka.persistence.SnapshotSelectionCriteria
+import cats.syntax.all._
+import cats.{Applicative, FlatMap, ~>}
+import com.evolutiongaming.akkaeffect.Fail
+import com.evolutiongaming.catshelper.{Log, MeasureDuration, MonadThrowable}
+
+import java.time.Instant
+
+/**
+  * Describes communication with underlying snapshot storage
+  *
+  * @tparam A - snapshot
+  */
+trait Snapshotter[F[_], -A] {
+
+  /**
+    * Saves a `snapshot` of this snapshotter' state.
+    *
+    * @return outer F[_] is about saving in background, inner F[_] is about saving completed
+    */
+  def save(seqNr: SeqNr, snapshot: A): F[F[Instant]]
+
+  /**
+    * Deletes the snapshot identified by `sequenceNr`.
+    *
+    * @return outer F[_] is about deletion in background, inner F[_] is about deletion being completed
+    */
+  def delete(seqNr: SeqNr): F[F[Unit]]
+
+  /**
+    * @see [[akka.persistence.Snapshotter.deleteSnapshots]]
+    * @return outer F[_] is about deletion in background, inner F[_] is about deletion being completed
+    */
+  @deprecated(
+    "use Snapshotter.delete(Snapshotter.Criteria) instead",
+    "01-12-2023"
+  )
+  def delete(criteria: SnapshotSelectionCriteria): F[F[Unit]]
+
+  /**
+    * Deletes all snapshots matching `criteria`.
+    *
+    * @return outer F[_] is about deletion in background, inner F[_] is about deletion being completed
+    */
+  def delete(criteria: Snapshotter.Criteria): F[F[Unit]]
+}
+
+object Snapshotter {
+
+  final case class Criteria(maxSequenceNr: Long = Long.MaxValue,
```

Comment:
formatting?

Reply:
I've fixed this case but bare in mind that I've added `scalafmt` due possibility of using it from VS Code. I will not reformat existing/unchanged files, but feel right to format any code that was moved from module to module